### ISSUE TITLE
Possibility to define multiple values for a custom dimension

### ISSUE THREAD

**Issue by mattab** [matomo-org/matomo#16156]
For example when using Custom Dimensions of scope "Page" as a way to track categories of an article, often an article can belong to multiple categories. Therefore it would be useful to be able to track multiple values for a same dimension. eg. the ability to define an array of values instead of just a string.

**Comment by raDiesle** [comment]
As far as I understand, right now it is not supported to provide multiple values for Custom Dimensions and it is not planned right now, correct?

**Comment by tsteur** [comment]
Yes @raDiesle it's not planned yet unfortunately. Maybe custom variables would help meanwhile? It's otherwise hard to recommend anything as it might depend on the use case. would maybe using multiple dimensions help like `userRole1`, `userRole2`, etc? This might work if only segmenting is needed. Grouping by it be probably not possible though.

**Comment by raDiesle** [comment]
Thanks for quick reply! 
In my special case, right now it is sufficient to resolve by your suggested solution - so thanks for verification.
In the future, for my specific application, there will be user-groups & business-groups, where a user can have n-m combinations of a rule. The Cloud solution supports 25 custom dimensions, so maybe it will fit in.

**Comment by jayhoogle** [comment]
This is a must-have for our analytics setup - we track impressions of content that can often have multiple authors, something we need to keep track of for the correct distribution of royalties. Without support for multiple values we have to manually search our CMS for the authors which is time-consuming and should be unnecessary. Are you planning to implement this feature?

**Comment by Chardonneaur** [comment]
Hello,
I also have this need for a client of mine.
This is definitely the use case that @mattab described, one page can have multiple tags associated to it, for example, a blog post can be assigned to sport and to the taxonomy culture and to the taxonomy personality and to the taxonomy leisure.
1) If I set this as different custom dimensions, I will be stuck when creating a custom report as it can only go up to 3 dimensions. And nothing is saying that in one case it may go into slot 1, and in the other case into slot 2.
2) If I send all those values like this:
  _paq.push(['setCustomDimension', 1, 'Culture']);
  _paq.push(['setCustomDimension', 1, 'Sport']);
  _paq.push(['setCustomDimension', 1, 'Transport']);
  _paq.push(['trackPageView']);
it won't work neither as the last custom dimension is the only one which will be taken into consideration
3) If I use events like this:
  _paq.push(['trackPageView']);
  _paq.push(['trackEvent', 'Pageview', 'Competence', 'Culture']);
  _paq.push(['trackEvent', 'Pageview', 'Competence', 'Sport']);
  _paq.push(['trackEvent', 'Pageview', 'Competence', 'Transport']);
then the bounce rate will go to 0% which would be an unexpected behaviour.

-> Is there then really a need to discontinue the custom variables? Shouldn't it be renamed? Because as far as I know, this is also useful for custom search engine results.

**Comment by justinberedo** [comment]
Hi, 

A client of ours needs this functionality. The purpose is for tracking multiple roles for an individual. For now, the custom variable plugin has this feature. But it's sad to see that support is to be dropped. Could we request having this functionality for custom dimensions? I think it would be very useful.

Thanks!

**Comment by heurteph-ei** [comment]
Why not tracking a dimension with value: `Culture|Sport|Leisure`.
Then when using segments, you can ask for the dimension containing Sport?

Or also if you really needs this feature, don't hesitate to sponsor Matomo...

**Comment by JiveDig** [comment]
+1 on needing this, for very similar use-cases as others, primarily posts in multiple categories.

**Comment by Starker3** [comment]
We have another user requesting this feature today. Would be great to have this implemented.
Their use case would be to track the User roles for a logged in user where there are multiple different roles that a user can have, which means there can be many different combinations of roles.

**Comment by MrJibus** [comment]
+1, we have a use case when we need a custom dimension with multiple values.

**Comment by VaDominikKienzler** [comment]
+1 very important feature

**Comment by peterbo** [comment]
+1
